packages/datadog-instrumentations/src/child_process.js

```diff
@@ -0,0 +1,171 @@
+'use strict'
+
+const util = require('util')
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const childProcessChannelStart = channel('datadog:child_process:execution:start')
+const childProcessChannelFinish = channel('datadog:child_process:execution:finish')
+const childProcessChannelError = channel('datadog:child_process:execution:error')
+
+// ignored exec method because it calls to execFile directly
+const execAsyncMethods = ['execFile', 'spawn']
+const execSyncMethods = ['execFileSync', 'spawnSync']
+
+const names = ['child_process', 'node:child_process']
+
+// child_process and node:child_process returns the same object instance, we only want to add hooks once
+let patched = false
+names.forEach(name => {
+  addHook({ name }, childProcess => {
+    if (!patched) {
+      patched = true
+      shimmer.massWrap(childProcess, execAsyncMethods, wrapChildProcessAsyncMethod())
+      shimmer.massWrap(childProcess, execSyncMethods, wrapChildProcessSyncMethod())
+      shimmer.wrap(childProcess, 'execSync', wrapChildProcessSyncMethod(true))
+    }
+
+    return childProcess
+  })
+})
+
+function normalizeArgs (args) {
+  const childProcessInfo = {
+    command: args[0]
+  }
+
+  if (Array.isArray(args[1])) {
+    childProcessInfo.command = childProcessInfo.command + ' ' + args[1].join(' ')
+    if (args[2] != null && typeof args[2] === 'object') {
+      childProcessInfo.options = args[2]
+    }
+  } else if (args[1] != null && typeof args[1] === 'object') {
+    childProcessInfo.options = args[1]
+  }
+
+  return childProcessInfo
+}
+
+function wrapChildProcessSyncMethod (shell = false) {
+  return function wrapMethod (childProcessMethod) {
+    return function () {
+      if (!childProcessChannelStart.hasSubscribers || arguments.length === 0) {
+        return childProcessMethod.apply(this, arguments)
+      }
+
+      const childProcessInfo = normalizeArgs(arguments)
+
+      if (shell || childProcessInfo.options?.shell === true || typeof childProcessInfo?.options?.shell === 'string') {
+        childProcessInfo.shell = true
+      } else {
+        childProcessInfo.shell = false
+      }
+
+      childProcessChannelStart.publish({ command: childProcessInfo.command, shell: childProcessInfo.shell })
+
+      let error
+      let result
+      try {
+        result = childProcessMethod.apply(this, arguments)
+        return result
+      } catch (err) {
+        childProcessChannelError.publish(err)
+        error = err
+        throw err
+      } finally {
+        const exitCode = error?.status || error?.code || result?.status || 0
+        if (!error && (exitCode !== 0 || result?.error)) {
+          childProcessChannelError.publish()
+        }
+        childProcessChannelFinish.publish({ exitCode })
+      }
```

Comment:
Why not use tracing channel here?

Reply:
I don't wanna answer for Julio but IMO we have a lesser understanding of tracing channel. Maybe we need a little training on it ? I know I do.

Reply:
Feel free to put something on my calendar to explain it if you feel you need that. 😄

Reply:
I've always forget about them. The code looks better at least for the `promisify` and synchronous methods. However for the asynchronous ones, since the return an `eventEmitter` I left the original code, I tried to wrap the `eventEmitter` with a promise to use `tracePromise` which worked but it kinda looked awkward.